core/adminserver/admin_server.go

```diff
@@ -0,0 +1,220 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package adminserver contains the KeyTransparencyAdminService implementation
+package adminserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/keytransparency/core/adminstorage"
+	"github.com/google/keytransparency/core/crypto/vrf/p256"
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/google/trillian/crypto/keys/der"
+	"github.com/google/trillian/crypto/keyspb"
+	"github.com/google/trillian/crypto/sigpb"
+
+	"github.com/golang/protobuf/ptypes"
+
+	google_protobuf "github.com/golang/protobuf/ptypes/empty"
+	gpb "github.com/google/keytransparency/core/proto/keytransparency_v1_grpc"
+	pb "github.com/google/keytransparency/core/proto/keytransparency_v1_proto"
+)
+
+var (
+	vrfKeySpec = &keyspb.Specification{
+		Params: &keyspb.Specification_EcdsaParams{
+			EcdsaParams: &keyspb.Specification_ECDSA{
+				Curve: keyspb.Specification_ECDSA_P256,
+			},
+		},
+	}
+	logArgs = &trillian.CreateTreeRequest{
+		Tree: &trillian.Tree{
+			DisplayName:        "KT SMH Log",
+			TreeState:          trillian.TreeState_ACTIVE,
+			TreeType:           trillian.TreeType_LOG,
+			HashStrategy:       trillian.HashStrategy_OBJECT_RFC6962_SHA256,
+			SignatureAlgorithm: sigpb.DigitallySigned_ECDSA,
+			HashAlgorithm:      sigpb.DigitallySigned_SHA256,
+			MaxRootDuration:    ptypes.DurationProto(0 * time.Millisecond),
+		},
+		KeySpec: &keyspb.Specification{
+			Params: &keyspb.Specification_EcdsaParams{
+				EcdsaParams: &keyspb.Specification_ECDSA{
+					Curve: keyspb.Specification_ECDSA_P256,
+				},
+			},
+		},
+	}
+	mapArgs = &trillian.CreateTreeRequest{
+		Tree: &trillian.Tree{
+			DisplayName:        "KT Map",
+			TreeState:          trillian.TreeState_ACTIVE,
+			TreeType:           trillian.TreeType_MAP,
+			HashStrategy:       trillian.HashStrategy_CONIKS_SHA512_256,
+			SignatureAlgorithm: sigpb.DigitallySigned_ECDSA,
+			HashAlgorithm:      sigpb.DigitallySigned_SHA256,
+			MaxRootDuration:    ptypes.DurationProto(0 * time.Millisecond),
+		},
+		KeySpec: &keyspb.Specification{
+			Params: &keyspb.Specification_EcdsaParams{
+				EcdsaParams: &keyspb.Specification_ECDSA{
+					Curve: keyspb.Specification_ECDSA_P256,
+				},
+			},
+		},
+	}
+)
+
+type server struct {
+	storage adminstorage.Storage
+	client  trillian.TrillianAdminClient
+	keygen  keys.ProtoGenerator
+}
+
+// New returns a KeyTransparencyAdminService implementation.
+func New(
```

Comment:
Please combine the next few lines into one, or only two lines. I don't think this is very readable. Thanks.

Reply:
Done.  Also renamed `trillian` to `tpb` to make the line shorter.